models/standardized_models/stripe__line_item_enhanced.sql

```diff
@@ -21,6 +21,38 @@ with invoice_line_item as (
     select *
     from {{ ref('stg_stripe__subscription') }}  
 
+), subscription_item as (
+
+    select *
+    from {{ ref('stg_stripe__subscription_item') }}  
+
+/*
+current_period_start/end fields currently exist in both subscription_item and subscription tables. Source depends on customer version of Stripe API
+current_period_start/end fields are deprecated from subscription table in most recent API versions so both sources are needed
+subscription_item allows for one-to-many relationships between invoice_id and plan, causing duplication at the line item level 
+select distinct on subscription_id to handle duplication and prioritize current_period_start/end from subscription_item table
+*/  
+
+), subscription_item_deduped as (
+
+    select distinct (subscription_id)
+        subscription_id,
+        source_relation,
+        current_period_start,
+        current_period_end
+    from subscription_item
```

Comment:
```suggestion
    select
        subscription_id,
        source_relation,
        min(current_period_start) as current_period_start,
        max(current_period_end) as current_period_end
        
    from subscription_item
    group by 1,2
```

This is _personally_ how I would do it but very open to other opinions! We shouldn't need to min/max `current_period_start` and `current_period_end`, but this could be a good sanity check/guardrail to ensure fanout can never happen, since we're aggregating to the subscription_id/source_relation level